### PR TITLE
Fix the link for Modern C

### PR DESCRIPTION
**Pull request by LipiLee**
The link is not valid for it.

## What does this PR do?
Add resource(s) | Remove resource(s) | Add info | Improve repo

## Checklist:
- [ ] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md).
- [ ] [Search](https://ebookfoundation.github.io/free-programming-books-search/) for duplicates.
- [ ] Include author(s) and platform where appropriate.
- [ ] Put lists in alphabetical order, correct spacing.
- [ ] Add needed indications (PDF, access notes, under construction).
- [ ] Used an informative name for this pull request.

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

File: books/free-programming-books-langs.md [hunk 1]
```diff
@@ -445,7 +445,7 @@ That section got so big, we decided to split it into its own file, the [BY SUBJE
 * [Learning GNU C](https://download-mirror.savannah.gnu.org/releases/c-prog-book/learning_gnu_c.pdf) - Ciaran O’Riordan (PDF)
 * [Let us C](https://web.archive.org/web/20211006163041/http://pdvpmtasgaon.edu.in/uploads/dptcomputer/Let%20us%20c%20-%20yashwantkanetkar.pdf) - Yashavant Kanetkar (PDF) *(:card_file_box: archived)*
 * [Modeling with Data](https://ben.klemens.org/pdfs/gsl_stats.pdf) - Ben Klemens (PDF)
-* [Modern C](https://modernc.gforge.inria.fr) - Jens Gustedt (PDF)
+* [Modern C](https://hal.inria.fr/hal-02383654/document) - Jens Gustedt (PDF)
```
Comment (davorpa) [inline books/free-programming-books-langs.md:448]: Should the http version the right link? It works

http://modernc.gforge.inria.fr/

There are other downloadable content like examples there! Of course the PDF where https://hal.inria.fr/hal-02383654/document points to

**Comment by davorpa** [comment]
Related to #6942

File: books/free-programming-books-langs.md (hunk 1 above)
Comment (LipiLee) [reply books/free-programming-books-langs.md:448]: The link didn't work at my first visit. So I've found its PDF version and update it. But, it works now.

**Comment by eshellman** [comment]
Thanks, and welcome to free-programming-books!

**Pull request merged**